pkg/resources/resource_cluster.go

```diff
@@ -74,6 +75,13 @@ var clusterSchema = map[string]*schema.Schema{
 			},
 		},
 	},
+	"use_name_as_id": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Use the cluster name as the Terraform resource ID instead of the internal cluster ID.",
+		ForceNew:    true,
```

Comment:
You probably _could_ handle in-place changes to this parameter, no? Just entails updating the resource ID.